File: SOLID/DependencyInversion/domain/booking/BookingService.cs
namespace SOLID.DependencyInversion.domain.booking
{
    public class BookingService
    {
        private readonly IAvailabilityDao bookings;

        public BookingService(IAvailabilityDao bookings)
        {
            this.bookings = bookings;
        }

        public BookingOutcome Book()
        {
            bool successful = bookings.IsAvailable();
            return new BookingOutcome(successful);
        }
    }
}

File: Java/Exercices/workspaces/solid-dip-start/src/main/java/zenika/solid/dip/domain/booking/BookingService.java
package zenika.solid.dip.domain.booking;

import zenika.solid.dip.infrastructure.AvailabilityDao;

public class BookingService {
    
    private final AvailabilityDao bookings;

    public BookingService(AvailabilityDao bookings) {
        this.bookings = bookings;
    }

    public BookingOutcome book() {
        boolean successful = bookings.isAvailable();
        return new BookingOutcome(successful);
    }
    
}
